### PR TITLE
Additional token fields. Edition search.

### PR DESCRIPTION
**Pull request by andygray**

### CODE REVIEW

File: build/schema.graphql [hunk 1]
```diff
@@ -217,3 +217,35 @@ type Collector @entity {
...
+    totalSupply: BigInt!
+    totalAvailable: BigInt!
+    active: Boolean!
+    name: String
+    description: String
+    image: String
+    tagstring: String
+    artist: String
+    artistAccount: Bytes! # address
+}
```
Comment (andygray) [inline build/schema.graphql:251]: ideally I would like a combined `KODA` entity that is a searchable edition and/or token

File: src/mappings/known-origin-v2-mapping.ts [hunk 2]
```diff
@@ -49,10 +49,14 @@ export function handleEditionCreated(event: EditionCreated): void {
     let editionEntity = loadOrCreateEdition(event.params._editionNumber, event.block, contract)
     editionEntity.save()
 
+    // FIXME temp
```
Comment (jamesmorgan) [inline src/mappings/known-origin-v2-mapping.ts:52]: ?

File: src/services/Edition.service.ts [hunk 3]
```diff
@@ -72,8 +72,50 @@ export function loadEdition(editionNumber: BigInt): Edition | null {
...
+
+            let metaData = constructMetaData(_editionData.value7)
+
+            editionFlatEntity.name = metaData.name
+            editionFlatEntity.description = metaData.description
+            editionFlatEntity.image = metaData.image
+            editionFlatEntity.artist = metaData.artist
+            editionFlatEntity.artistAccount = getArtistAddress(_editionData.value4)
+
+            editionFlatEntity.tagstring = metaData.tags.toString()
```
Comment (jamesmorgan) [inline src/services/Edition.service.ts:113]: I guess in the end you could just add the flat fields to `edition` entity - would also be good to do this for `token` eventually as well

Comment (andygray) [reply src/services/Edition.service.ts:113]: I see them all being flattened

File: src/mappings/known-origin-v2-mapping.ts (hunk 2 above)
Comment (andygray) [reply src/mappings/known-origin-v2-mapping.ts:52]: I am actually going to improve this entity

File: src/services/MetaData.service.ts [hunk 4]
```diff
@@ -55,11 +55,16 @@ export function constructMetaData(tokenURI: string): MetaData | null {
...
+            log.error("Unknown IPFS data found for token URI {}", [tokenURI]);
+        }
+
+        return metaData;
     } else {
         log.error("Unknown IPFS hash found for token URI {}", [tokenURI]);
     }
 
-    return metaData;
+    return null
```
Comment (jamesmorgan) [inline src/services/MetaData.service.ts:67]: Is this the bug? running `null` - `constructMetaData()` is called here - https://github.com/knownorigin/known-origin-subgraph/pull/8/files#diff-3b707fca3c0b544b060c217153880a73R37 - it wouldnt supprise me if there is a bad edition on Rinkeby with no IPFS data associated to it

**Pull request closed without merging**